snuba/web/db_query.py

```diff
@@ -0,0 +1,211 @@
+import logging
+
+from functools import partial
+from hashlib import md5
+from typing import (
+    Any,
+    Mapping,
+    MutableMapping,
+    Optional,
+)
+
+from snuba import settings, state
+from snuba.clickhouse.errors import ClickhouseError
+from snuba.clickhouse.query import ClickhouseQuery
+from snuba.environment import reader
+from snuba.redis import redis_client
+from snuba.request import Request
+from snuba.state.cache import Cache, RedisCache
+from snuba.state.rate_limit import (
+    PROJECT_RATE_LIMIT_NAME,
+    RateLimitAggregator,
+    RateLimitExceeded,
+)
+from snuba.util import force_bytes
+from snuba.utils.codecs import JSONCodec
+from snuba.utils.metrics.timer import Timer
+from snuba.web import RawQueryException, RawQueryResult
+from snuba.web.query_metadata import ClickhouseQueryMetadata, SnubaQueryMetadata
+
+cache: Cache[Any] = RedisCache(redis_client, "snuba-query-cache:", JSONCodec())
+
+logger = logging.getLogger("snuba.query")
+
+
+def update_query_metadata_and_stats(
+    request: Request,
+    sql: str,
+    timer: Timer,
+    stats: MutableMapping[str, Any],
+    query_metadata: SnubaQueryMetadata,
+    query_settings: Mapping[str, Any],
+    trace_id: Optional[str],
+    status: str,
+) -> MutableMapping[str, Any]:
+    """
+    If query logging is enabled then logs details about the query and its status, as
+    well as timing information.
+    Also updates stats with any relevant information and returns the updated dict.
+    """
+    stats.update(query_settings)
+
+    query_metadata.query_list.append(
+        ClickhouseQueryMetadata(sql=sql, stats=stats, status=status, trace_id=trace_id)
+    )
+
+    return stats
+
+
+def raw_query(
```

Comment:
I made almost no change to this method. Just copied over to a separate file and removed the dataset dependency.

Reply:
Sorry my comment was not accurate. I wanted to keep what depends on dataset in a file and what does not in another. There was no dependency from this specific  function.